functions/src/application/portal.ts

```diff
@@ -302,3 +303,31 @@ export const get_developer_profile = async (request: Request, response: Response
     });
   }
 };
+
+export const create_vanity_link = async (req: Request, res: Response) => {
+  const { first_name, last_name, email, destination, primary_domain, subdomain, slashtag } = req.body;
+  let error = null;
+
+  await build_vanity_link_v2({
+    first_name,
+    last_name,
+    email,
+    destination,
+    primary_domain,
+    subdomain,
+    slashtag,
+  }).catch((err) => {
+    error = [err.response.data];
+  });
+
+  if (!error)
+    return res.status(201).send({
+      message: "Sucessfully created Vanity link",
+      url: `https://${subdomain}.${primary_domain}/${slashtag}`,
+    });
+
+  return res.status(400).send({
+    message: "Failed to create Vanity link",
+    error,
+  });
```

Comment:
Minor nit: reverse the operations here by switching the `if (!error)` condition with the equivalent one for `if (error)`. We normally check for an error first since we usually expect the code to execute without failure.